client/src/containers/PostDashboard.tsx

```diff
@@ -5,26 +5,37 @@ import Container from '@mui/material/Container';
 import Typography from '@mui/material/Typography';
 import Pagination from '@mui/material/Pagination';
 
+import { User } from 'models/user';
+
 import Header from '../components/Header';
 import PostPreview from '../components/PostPreview';
-
-import { api_v1 } from '../api/v1';
 import CreatePost from '../components/CreatePost';
 
+import ServerApi, { APIResponse, PostUser } from '../api/v1';
+import { useNavigate } from 'react-router-dom';
+
+const api = new ServerApi();
+
+export const UserContext: React.Context<{
```

Comment:
But the user will not be authenticated on App.tsx, we only want to get the user after authentication (which will route them to the dashboard). The rest of our site is contained on the dashboard

Reply:
It could be an empty object. The typical pattern is to have user data in something like redux which is app wide state.

I feel like there would be a need to make a change in the future, but I'll leave it up to you.